app/Models/User.php

```diff
@@ -68,6 +69,11 @@ class User extends Authenticatable implements WebAuthnAuthenticatable
 		'created_at',
 		'updated_at',
 		'token',
+
+		// We do not forward those anymore to the front end.
```

Comment:
```php
		 * We do not forward those to the front end: they are provided by {@link \App\DTO\UserWithCapabilitiesDTO}.
		 * We do not need to inform every user on Lychee who can upload etc.
```

Reply:
That's a point which I would like to discuss. I would not do that. IMHO, this only adds another layer of complexity which only provides little to none security advantage. The security of Lychee should not depend on the fact whether some user can learn whether some other user has the upload capability or not.

However, it makes the code much more complicated on the backend, but in particular on the frontend. If we have an object of a certain class, then this object should always have the same attributes. (This will be even more important, when I will eventually continue on my Android client which uses a strongly typed language such as Java.) But also for a prototype-based language such as JavaScript, i would not like to need permanently thinking about whether one can safely use those attributes of a specific user object or if they might be undefined.

IMHO, this is some kind of micro-optimization whose costs will outweigh the (IMHO non-existing) security benefit in the long run, because this approach of making every response slightly inconsistent to another one has the potential to shoot into our feet, but at least it will become a PITA.